ChronoStore/src/storywriter.cpp

```diff
@@ -0,0 +1,497 @@
+            if(attributeDataBuffer != NULL) free(attributeDataBuffer);
+
+            if(attr >=0)    H5Sclose(attr);
+            if(attr1 >=0)    H5Aclose(attr1);
+            if(attributeType >= 0)  H5Tclose(attributeType);
+            if(chunkmetaAttribute >=0 ) H5Tclose(chunkmetaAttribute);
+            if(eventType >=0 )  H5Tclose(eventType);
+            if(storyDataset >= 0)   H5Dclose(storyDataset);
+            if(ChronicleH5 >= 0)  H5Fclose(ChronicleH5);
+            return -1;
+        }
+
+        // Release extendAttributeBuffer memory
+        free(extendChunkMetadataBuffer);
+    }
+    else{
+        hsize_t attributeDims[] = {attributeCount};
+        attr = H5Screate(H5S_SIMPLE);
+        ret  = H5Sset_extent_simple(attr, ATTRIBUTE_CHUNKMETADATA_RANK, attributeDims, NULL);
+
+        /*
+        * Create array attribute and write array attribute.
+        */
+        attr1 = H5Acreate2(storyDataset, ATTRIBUTE_CHUNKMETADATA, chunkmetaAttribute, attr, H5P_DEFAULT, H5P_DEFAULT);
+        attrError = H5Awrite(attr1, chunkmetaAttribute, attributeDataBuffer);
+
+        /*
+        * Create and write to Min and Max and totalStoryEvents attribute.
+        */
+        attributeDims[0] = {1};
+        attr = H5Screate_simple(1, attributeDims, NULL);
+
+        attr_DatasetMin = H5Acreate2(storyDataset, ATTRIBUTE_DATASET_MIN, H5T_NATIVE_ULONG, attr, H5P_DEFAULT, H5P_DEFAULT);
+        attrMinError = H5Awrite(attr_DatasetMin, H5T_NATIVE_ULONG, &eventBuffer[0].timeStamp);
+
+        attr_DatasetMax = H5Acreate2(storyDataset, ATTRIBUTE_DATASET_MAX, H5T_NATIVE_ULONG, attr, H5P_DEFAULT, H5P_DEFAULT);
+        attrMaxError = H5Awrite(attr_DatasetMax, H5T_NATIVE_ULONG, &eventBuffer[storyChunkDims[0]-1].timeStamp);
+
+        attr_DatasetTotalStoryEvents = H5Acreate2(storyDataset, ATTRIBUTE_TOTAL_DATASET_STORY_EVENTS, H5T_NATIVE_ULONG, attr, H5P_DEFAULT, H5P_DEFAULT);
+        attrStoryEventsError = H5Awrite(attr_DatasetTotalStoryEvents, H5T_NATIVE_ULONG, &storyChunkDims[0]);
+        if(attrStoryEventsError < 0 || attr_DatasetTotalStoryEvents < 0 || attrMinError < 0 || attr_DatasetMin < 0 || attrMaxError < 0 || attr_DatasetMax < 0 || attr < 0 || attr1 < 0 || attrError < 0)
+        {
+            if(attr_DatasetTotalStoryEvents >= 0)   H5Aclose(attr_DatasetTotalStoryEvents);
+            if(attr_DatasetMax >= 0)    H5Aclose(attr_DatasetMax);
+            if(attr_DatasetMin >= 0)    H5Aclose(attr_DatasetMin);
+            if(attr1 >= 0)  H5Aclose(attr1);
+            if(attr >= 0)   H5Sclose(attr);
+            if(attributeDataBuffer != NULL) free(attributeDataBuffer);
+
+            if(chunkmetaAttribute >=0 ) H5Tclose(chunkmetaAttribute);
+            if(eventType >=0 )  H5Tclose(eventType);
+            if(storyDataset >= 0)   H5Dclose(storyDataset);
+            if(ChronicleH5 >= 0)  H5Fclose(ChronicleH5);
+            return 1;
+        }
+    }
+
+    /*
```

Comment:
May be a jump/goto to this point in the code where you release resources in case of errors would have improved readability.